chalice/awsclient.py

```diff
@@ -92,8 +108,12 @@ def create_function(self,
             kwargs['Timeout'] = timeout
         if memory_size is not None:
             kwargs['MemorySize'] = memory_size
-        return self._call_client_method_with_retries(
-            self._client('lambda').create_function, kwargs)['FunctionArn']
+        try:
+            return self._call_client_method_with_retries(
+                self._client('lambda').create_function, kwargs)['FunctionArn']
+        except Exception as e:
+            self._raise_large_deployment_related_errors(e, len(zip_contents))
```

Comment:
Consider creating some lambda specific error so pieces higher up the stack know that we failed in the lambda function.